### ISSUE TITLE
avoiding copies

### ISSUE THREAD

**Issue by stevengj** [JuliaPy/PyCallJLD.jl#2]
See the discussion in JuliaPy/PyCall.jl#388 about avoiding copies of buffers; maybe some code can be shared.

**Comment by cstjean** [comment]
Tim Holy wrote in the JLD README 2 years ago:

> For lossless storage of arbitrary Julia objects, the only other complete solution appears to be Julia's serializer, which can be accessed via the serialize and deserialize commands. However, because the serializer is also used for inter-process communication, long-term backwards compatibility is currently uncertain. (The JLDArchives package exists to test compatibility of older JLD file formats.) If you choose to save data using the serializer, please use the file extension .jls to distinguish the files from .jld files.

But apart from this comment is there any reason why I shouldn't just write

```julia
function JLD.writeas(pyo::PyObject)
    o = IOBuffer()
    serialize(o, pyo)
    PyObject_Serializer(takebuf_string(o))
end
```
assuming that Base.serialize is correctly implemented?

**Comment by stevengj** [comment]
I was thinking that you could call pickle directly to get a byte array, but that you could either copy the code from 388 or maybe we could refactor it slightly (though the code you would need to copy/adapt is just one or two lines).

(I also wouldn't use a `String` for this since the pickle output is really just arbitrary binary data, unrelated to UTF-8.  Just use a `Vector{UInt8}`?)

**Comment by cstjean** [comment]
Fixed by #3
